### PR TITLE
RendererDRMPRIME: release video buffers after flush

### PR DESCRIPTION
**Pull request by Kwiboo**
## Description
This PR fixes a leak of AVFrame (video buffers) and stalled video using RendererDRMPRIME.

## Motivation and Context
Fixes video buffer and video buffer pool leak after render manager makes a Flush call.
Also fixes stalled video after a Flush call or use of uninitialized iRepeatPicture.

~~Unusual nextFramePts can be seen in A/V timing debug logs even when video codec always set proper pts.
`nextFramePts: 5714210702664009105337883776014638214130495562162101884065277216588220643083327355436890770015766941100908080523945603259752984551726898686331406795657880409056094484939032126274458472512670320569807338447004032694277645696292734360182499080257922924544.000000`
`nextFramePts: -3429978449449829381157260441441042040285482044900332944707601209004516666209880432582333383265363956854653933465998316116411007460523207625177359482136722347997097235720548778890900365580714573089773857218111323294232445263896076015828865810779407894771138560000.000000`~~

~~`CVideoBuffer: initialize members` should probably fix same issue as seen in https://github.com/xbmc/xbmc/pull/13594~~

ping @lrusak @FernetMenta @popcornmix

## How Has This Been Tested?
Playing mpeg2 and other videos on my Rock64 + LibreELEC

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by FernetMenta** [comment]
> CVideoBuffer: initialize members should probably fix same issue as seen in #13594

All members of this class have to be set by decoder. Init members is useless. #13594 is wrong and my questions have not been answered on this PR.

**Comment by Kwiboo** [comment]
Somehow `m_Queue[].pts` ends up with bogus values as seen in `nextFramePts` values, VideoCodecDRMPRIME sets pts for all VideoPictures passing through so not sure from where the bogus values is coming from, I will add some logging to around https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp#L968 to see when bogus pts is set.
Maybe https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h#L197 should init to `DVD_NOPTS_VALUE` ?

**Comment by FernetMenta** [comment]
> https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h#L197 should init to DVD_NOPTS_VALUE ?

no. if nothing is in there, the value is irrelevant. if somthing is in, the value was set.

**Comment by popcornmix** [comment]
@Kwiboo I had problems with uninitalised elements of m_picture causing spurious timestamps that sometimes stalled video at start of playback. See: https://github.com/xbmc/xbmc/pull/13594
(the initial version looked very much like your initialisation commit).

I accept that may not be the right solution - perhaps mmal (and presumably DRMPrime too) may not be initialising something when populating pictures, but I haven't tracked it down.

**Comment by Kwiboo** [comment]
@popcornmix that sound like the exact same issue I have observed with the drm prime decoder, I will initialize pts in VideoPicture and SPresent to poison values and look where it gets used, to my knowledge the drm prime decoder should set proper pts values on each VideoPicture.

**Comment by Kwiboo** [comment]
@FernetMenta @popcornmix the spurious timestamps / stalled video at start of playback was caused by `iRepeatPicture` not being set in `GetPicure`

**Comment by popcornmix** [comment]
@Kwiboo thanks - that does sound plausible.

**Comment by FernetMenta** [comment]
good to hear that you tracked it down

**Comment by Kwiboo** [comment]
updated to include a small fix for `color_range`

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp [hunk 1]
```diff
@@ -78,6 +78,8 @@ bool CRendererDRMPRIME::Configure(const VideoPicture& picture, float fps, unsign
 void CRendererDRMPRIME::AddVideoPicture(const VideoPicture& picture, int index, double currentClock)
 {
   BUFFER& buf = m_buffers[index];
+  if (buf.videoBuffer)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp]: you should log an error here. LinuxRendererGL (reference) does not have this check. Means if it ever releases a buffer here, something went really wrong.

Comment (Kwiboo) [reply xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp]: this is a workaround and we expect to release video buffers here after a call has been made to `Flush`, if a video buffer is released and removed in `Flush` while it is tied to the primary drm plane the entire drm display pipeline can be disabled.

a proper solution where the video buffers is reference counted while being displayed on screen (tied to a drm plane) will follow later, may require some refactoring and hooks into gbm windowing

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp]: maybe place a comment here why you do this

Comment (Kwiboo) [reply xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp]: a short comment has been added

**Review by lrusak — APPROVED** [review]

**Pull request merged**